flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -588,8 +588,8 @@ private void startTriggeringCheckpoint(
 			final CompletableFuture<?> coordinatorCheckpointsComplete = pendingCheckpointCompletableFuture
 					.thenComposeAsync((pendingCheckpoint) ->
 							OperatorCoordinatorCheckpoints.triggerAndAcknowledgeAllCoordinatorCheckpointsWithCompletion(
-									coordinatorsToCheckpoint, pendingCheckpoint, timer),
-							timer);
+									coordinatorsToCheckpoint, pendingCheckpoint, mainThreadExecutor),
```

Comment:
Shouldn't this commit:
> [hotfix] Fix broken DefaultSchedulerTest by satisfying the main thread requirement of CheckpointCoordinator

be squashed with the preceding one?

Reply:
Oops, I just realize this commit includes some codes belongs the preceding commit. It's must caused by the conflict resolving. And yes, you are right. This commit could be squashed with the preceding one. I will do the squashing when all comments are addressed.